tests/static_analyzer/test_cpp_adapter.py

```
@@ -0,0 +1,246 @@
+
+
+class TestCompilationDatabaseGuard:
+    """``get_lsp_command`` must reject projects without a compile DB.
+
+    Without a ``compile_commands.json`` / ``compile_flags.txt``, clangd
+    silently returns empty cross-file references. Fail fast with actionable
+    install instructions instead.
+    """
+
+    def test_raises_when_no_compilation_database(self, tmp_path: Path) -> None:
+        with pytest.raises(RuntimeError, match="compile_commands.json"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)